src/service/service_layer.cc

```diff
@@ -100,6 +105,35 @@ void ServiceLayer::ReadDfs(const std::string& key_base,
   ReadDfs(key_base, chirps, counter + 1);
 }
 
+std::vector<ChirpObj> ServiceLayer::HashTag(const std::string& uname,const std::string& hash_tag) {
+  //getting all users
+  std::vector<std::string> users = ds_.Get(AllUserKey_); 
+  for (const std::string& u : users) {
+    PutMonitorKey(uname, u);
+  }
+
+  std::vector<ChirpObj> chirps;
+  std::string monitor_check_base = uname + kMonitorCheckKey_;
```

Comment:
Why do you use monitor bookkeeping info?

Reply:
I was little confused about the Key and Value you stored in KV store for monitor. So I keep using the previous code in Monitor for Stream. I am going to fix it.

Reply:
Do you have a function checking if a user is register?

Reply:
you can just check if the username exists as a key in the store